src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs

```diff
@@ -1468,13 +1469,16 @@ private string ReadLineFromConsole(bool endOnTab, string initialContent, bool ca
             _rawui.ClearKeyCache();
             uint keyState = 0;
             string s = string.Empty;
+            Span<char> inputBuffer = stackalloc char[MaxInputLineLength + 1];
+            initialContent.AsSpan().CopyTo(inputBuffer);
+
 #endif
                 do
                 {
 #if UNIX
                     keyInfo = Console.ReadKey(true);
 #else
-                s += ConsoleControl.ReadConsole(handle, initialContent, maxInputLineLength, endOnTab, out keyState);
+                s += ConsoleControl.ReadConsole(handle, initialContent.Length, inputBuffer, MaxInputLineLength, endOnTab, out keyState);
```

Comment:
`initialContent`  is actually `lastInput` in ReadLineWithTabCompletion()
Next comment is related. We can discuss there.

Reply:
My bad. I misunderstood the behavior of `CONSOLE_READCONSOLE_CONTROL.nInitialChars`.

Reply:
I stumbled at the same place :-)

It would be good to split Windows and Unix code paths and make separate methods under `#if` - this would make the code more readable (in follow PR). What do you think?

Reply:
I didn't look closely, but yes, the code looks pretty ugly even with a glance :)
It would be great if you have time to look into how to better organize the win/unix code.